### PR TITLE
Fix milestone reminder

### LINKED ISSUE #42835: Milestone Reminder GH Action doesn't always work
### Describe the bug

We have a very useful milestone reminder GH action, but it doesn't always work as expected.
There are two scenarios we need to fix:
1. The action sometimes literally fails because of an exception ([example](https://github.com/metabase/metabase/actions/runs/9129775294))
2. The action posting a comment on PRs that don't even have the closing keyword ([example](https://github.com/metabase/metabase/pull/42624#issuecomment-2110361435))

### To Reproduce

1. Close a PR that contains this body: `Closes #12345 and #23456`

or 

2. Close a PR that doesn't have a closing keyword at all.

### Expected behavior

1. Action should never fail
2. Action should not post comments on PRs that didn't have the closing keyword and a linked issue to begin with

### Logs

_No response_

### Information about your Metabase installation

```JSON
.
```

### Severity

p3

### Additional context

Internal.

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Description

Resolves #42835.

The milestone reminder workflow has been broken.
My previous attempt to fix it didn't really work because the parsing logic was wrong.

This is the example:
https://github.com/metabase/metabase/actions/runs/9382661958/job/25834555041#step:2:78

Notice how it tries to fetch `https://api.github.com/repos/metabase/metabase/issues/s`

Some local experimentation explains why this happened in the past:
image

I've now included a lot of unit tests to make sure this script is working.

### Notable Changes
- The script is now able to parse multiple linked issues, which wasn't the case before
- The issue number with be included in the bot message ([check this comment](https://github.com/metabase/metabase/pull/43684#issuecomment-2150902966)) 

### How to verify
- Run accompanied unit tests
- Check this test PR out: https://github.com/metabase/metabase/pull/43684
    - It links to one issue with and one without the milestone, and the bot correctly parses the issue without the milestone

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: .github/workflows/check-milestone.yml [hunk 1]
```diff
@@ -19,17 +19,12 @@ jobs:
       - uses: actions/github-script@v7
         with:
           script: | # js
+            const { getLinkedIssues } = require('${{ github.workspace }}/release/utils/index.cjs');
```
Comment (iethree) [inline .github/workflows/check-milestone.yml:22]: js/cjs extension mismatch here

Comment (nemanjaglumac) [reply .github/workflows/check-milestone.yml:22]: I think I tried js in my fork and it didn't work out of the box.

File: .github/workflows/check-milestone.yml [hunk 2]
```diff
@@ -47,34 +46,41 @@ jobs:
...
+                    });
 
-            await github.rest.issues.createComment({
-              owner,
-              repo,
-              issue_number: pullNumber,
-              body: message,
-            });
+                    // Exit as soon as we found an issue without a milestone and alerted the author.
+                    process.exit(0);
```
Comment (nemanjaglumac) [inline .github/workflows/check-milestone.yml:83]: @iethree as the alternative, we can let the script go through ALL linked issues and alert for each unlinked issue individually. WDYT?

Comment (iethree) [reply .github/workflows/check-milestone.yml:83]: i think one message is sufficient, i like how you did it here 👍

**Review by iethree — APPROVED** [review]
💖

**Pull request merged**